scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -636,10 +637,29 @@
    Further limit the considerable jobs to a maximum of num-considerable jobs."
   [db pending-jobs user->quota user->usage num-considerable pool-name]
   (log/debug "In" pool-name "pool, there are" (count pending-jobs) "pending jobs:" pending-jobs)
-  (->> pending-jobs
-       (filter-based-on-quota user->quota user->usage)
-       (filter (fn [job] (util/job-allowed-to-start? db job)))
-       (take num-considerable)))
+  (let [enforcing-job-launch-rate-limit? (ratelimit/enforce? ratelimit/job-launch-rate-limiter)
+        user->number-jobs (atom {})
+        user->max-time-until-out-of-debt (atom {})
+        user-within-launch-rate-limit?-fn
+        (fn
+          [{:keys [job/user]}]
+          (let [time-until-out-of-debt-millis (ratelimit/time-until-out-of-debt-millis! ratelimit/job-launch-rate-limiter user)
+                in-debt? (not (zero? time-until-out-of-debt-millis))]
+            (when in-debt?
+              (swap! user->max-time-until-out-of-debt update user #(max (or % 0) time-until-out-of-debt-millis))
+              (swap! user->number-jobs update user #(inc (or % 0))))
+            (not (and in-debt? enforcing-job-launch-rate-limit?))))
+        out
```

Comment:
How about `considerable-jobs` instead of `out`?

Reply:
out says exactly what it is going to be used for. considerable-jobs doesn't.

Reply:
How does `out` tell me what this is going to be used for? It seems to me that `out` reveals nothing about what the thing is, how it might be used, or its shape.